libcudacxx/include/cuda/std/__cccl/architecture.h

```diff
@@ -0,0 +1,44 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of libcu++, the C++ Standard Library for your entire system,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef __CCCL_ARCH_H
+#define __CCCL_ARCH_H
+
+// The header provides the following macros to determine the host architecture:
+//
+// _CCCL_ARCH(ARM64)     ARM64
+// _CCCL_ARCH(X86)       X86 both 32 and 64 bit
+// _CCCL_ARCH(X86_64)    X86 64 bit
+// _CCCL_ARCH(X86_32)    X86 64 bit
+// _CCCL_ARCH(64BIT)     Any 64 bit OS (supported by CUDA)
+// _CCCL_ARCH(32BIT)     Any 32 bit OS (supported by CUDA)
+
+// Determine the host compiler and its version
+#if defined(__aarch64__) || defined(_M_ARM64) || defined(_M_ARM64EC) /* Windows Emulation Compatible */
+#  define _CCCL_ARCH_ARM64_() 1
+#elif defined(_M_X64) || defined(_M_IX86) || defined(__amd64__) || defined(__x86_64__)
+#  define _CCCL_ARCH_X86_() 1
+#  if defined(_M_X64) || defined(__amd64__) || defined(__x86_64__)
+#    define _CCCL_ARCH_X86_64_() 1
+#  else
+#    define _CCCL_ARCH_X86_32_() 1
+#  endif
+#endif
+
+#if defined(__aarch64__) || defined(_M_ARM64) || defined(_M_ARM64EC) || defined(__amd64__) || defined(__x86_64__) \
+  || defined(_M_X64)
+#  define _CCCL_ARCH_64BIT_() 1
+#elif defined(_M_IX86)
+#  define _CCCL_ARCH_32BIT_() 1
+#endif
+
+#define _CCCL_ARCH(...) _CCCL_ARCH_##__VA_ARGS__##_()
```

Comment:
This wont work because we do not define the error case

e.g `_CCCL_ARCH(ARM64)` will try to evaluate `_CCCL_ARCH_ARM64()` which is not defined for anything other than ARM64 and will error out.

We need to properly define the 0 case for those macros